lib/module_importer.js

```diff
@@ -60,6 +72,21 @@ function readFirstFile(uri, filenames, cb, examinedFiles) {
 function makeImporter(eyeglass, sass, options, fallbackImporter) {
   var importedFiles = {};
   var root = options.root;
+  var allModulesCache = {};
+
+  function getModuleByName(moduleName, dir) {
+    var allModules = allModulesCache[dir];
+    if (!allModules) {
+      allModules = discover.all(dir, true).modules;
```

Comment:
This isn't compatible with `depth`, as the cache will be for `depth = 0` and not account for other calls to `scan`

1. remove depth
2. use `depth` of 0 as part of the `allModules` cache